### PR TITLE
cli: Add option for not printing progress bar

### PR DESCRIPTION
**Pull request by mwcarlis**
Some users don't want to have the progress bar because
it creates log spam.

### CODE REVIEW

File: cli/main.c [hunk 1]
```diff
@@ -1456,6 +1456,7 @@ static int fw_update(int argc, char **argv)
 		int force;
 		int set_boot_rw;
 	} cfg = {};
+	int no_progress_bar = 0;
```
Comment (lsgunth) [inline cli/main.c:1459]: This should be within the cfg struct above.

File: cli/main.c [hunk 2]
```diff
@@ -97,7 +97,7 @@ int switchtec_handler(const char *optarg, void *value_addr,
  * build is non-trivial. After evaluating the development effort
  * and the resources available, we have decided to remove Windows
  * build support from release roadmap.
- * 
+ *
```
Comment (lsgunth) [inline cli/main.c:100]: Please avoid unrelated whitespace changes.

File: cli/main.c [hunk 3]
```diff
@@ -1528,8 +1531,13 @@ static int fw_update(int argc, char **argv)
 	}
 
 	progress_start();
-	ret = switchtec_fw_write_file(cfg.dev, cfg.fimg, cfg.dont_activate,
-				      cfg.force, progress_update);
+	if (no_progress_bar) {
```
Comment (lsgunth) [inline cli/main.c:1534]: No need for braces on this if statement.

File: cli/main.c [hunk 4]
```diff
@@ -1709,6 +1717,7 @@ static int fw_read(int argc, char **argv)
 		int bl2;
 		int key;
 	} cfg = {};
+	int no_progress_bar = 0;
```
Comment (lsgunth) [inline cli/main.c:1720]: As above

File: cli/main.c [hunk 5]
```diff
@@ -1770,9 +1781,14 @@ static int fw_read(int argc, char **argv)
 	}
 
 	progress_start();
-	ret = switchtec_fw_body_read_fd(cfg.dev, cfg.out_fd,
-					inf, progress_update);
-	progress_finish();
+	if (no_progress_bar) {
```
Comment (lsgunth) [inline cli/main.c:1784]: As above

**Comment by mwcarlis** [comment]
I'm not sure if it updated correctly here. I'm don't use github very often..

**Comment by kelvin-cao** [comment]
Thanks.

I tried to update the base branch to Microsemi:devel which is the target for PRs, but I realized your master branch was based on Microsemi:master and there're some commits not exist on devel which causes the merging not clean.

I can rebase devel and apply the changes in the background if you don't mind. 

@lsgunth I'm thinking we can change the default branch to 'devel' in github which will make 'devel' the default branch for cloning and PRs. What do you think?

**Comment by lsgunth** [comment]
No idea about the default branch, I guess you can try it. It does mean users will see that branch on the github home screen. I'm not sure if it actually affects cloning and PRs or not.

**Comment by mwcarlis** [comment]
I can also just update the change to match the branch you want.

**Comment by kelvin-cao** [comment]
I tried and the cloned repo does have the default branch changed. But I still think people should see master on this site.

@mwcarlis Could you rebase to devel and apply this change again? Sorry for the inconvenience.

**Comment by mwcarlis** [comment]
Hi sorry for delay. I just checked out devel and cherry-picked my change over from master, but it applied clean.

**Comment by kelvin-cao** [comment]
Yes, cherry pick should be clean as the conflicts are mostly the version commits. Could you update the from branch with the change and base branch in this PR? Not sure if you can update the from branch in this PR or use a new one.

**Comment by lsgunth** [comment]
Kelvin, maybe just do this yourself... You can checkout the branch on your command line and do a rebase and force push or just merge it to devel manually.

**Comment by kelvin-cao** [comment]
Yes, that'll be easier. Will do later when a recent release settled down.

**Comment by kelvin-cao** [comment]
Hi I've merged the changes. Commit# d3d24724dd8ee245688d7cf54224ed8f1c3c65ca.
